parlai/tasks/igc/agents.py

```diff
@@ -0,0 +1,339 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""
+Image Grounded Conversations (IGC) Task.
+
+See https://www.aclweb.org/anthology/I17-1047/ for more details. One must download
+the data from https://www.microsoft.com/en-us/download/details.aspx?id=55324
+prior to using this teacher.
+
+The images are then downloaded from the urls specified in the data. Unfortunately,
+not all links are live, and thus some examples do not have valid images.
+
+As there is no training set, we manually split 90% of the validation set
+into train.
+"""
+import csv
+import os
+
+from abc import ABC, abstractmethod
+from PIL import Image
+from typing import List, Dict, Any
+
+from parlai.core.build_data import download_multiprocess
+from parlai.core.params import Opt
+from parlai.core.teachers import AbstractImageTeacher
+
+import parlai.utils.typing as PT
```

Comment:
I would mind mirroring the regular typing. `From parlai.utils.typing import TShared`, but this is cool too. I'm happy.